src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -282,7 +282,19 @@ ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
     uct_rdmacm_cm_ep_t *cep = ucs_derived_of(ep, uct_rdmacm_cm_ep_t);
     char ip_port_str[UCS_SOCKADDR_STRING_LEN];
 
-    ucs_trace("rdma_disconnect on cm_id %p", cep->id);
+    ucs_trace("%s: rdma_disconnect on ep %p cm_id %p",
+              (cep->flags & UCT_RDMACM_CM_EP_ON_SERVER ? "server" : "client"),
+              cep, cep->id);
+
+    if (ucs_unlikely(cep->flags & UCT_RDMACM_CM_EP_DISCONNECT_CALL)) {
+        ucs_error("%s: uct_ep_disconnect should not be called more than once on the "
+                  "same endpoint (rdmacm ep=%p id=%p to peer %s)",
+                  (cep->flags & UCT_RDMACM_CM_EP_ON_SERVER ? "server" : "client"),
+                  cep, cep->id,
+                  ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                                   UCS_SOCKADDR_STRING_LEN));
+        return UCS_ERR_INVALID_PARAM;
```

Comment:
now think of it, probably better to return UCS_ERR_BUSY

Reply:
why busy? if the ep has already disconnected what do we mark as busy?

Reply:
if already disconnected - return some error
if started but not finished - return busy